### DISCUSSION TITLE
Add a field name to updated fields when renaming tiddlers?

### DISCUSSION THREAD

**Discussion by amreus** [TiddlyWiki/TiddlyWiki5#6957] [Ideas]
Is there currently, or could there be a user interface to add a field to the list of fields updated when renaming a tiddler?

For example, if I edit the title of [HelloThere](https://tiddlywiki.com/#HelloThere), there appears a checkbox asking to update the _**list**_ and _**tags**_ fields.  Is it possible to add our own list field to the list of fields to update?

Specifically, I have a _**parent**_ field I want to update if I change the title of tiddler.

**Comment by amreus** [comment]
For reference, it looks like the `tags` and `list` fields are hard-coded in `core/modules/wiki-bulk-ops.js`

[/core/wiki/bulkops.js](https://github.com/Jermolene/TiddlyWiki5/blob/dd66fcc759398f390a67c782f88ed8c517ed86bc/core/modules/wiki-bulkops.js#L47)

**Comment by amreus** [comment]
Also, `tags` and `list` are stored internally as Array types, where "normal" fields are strings. The relinking assumes the relink fields are arrays.

Is there a `$tw.util` to parse a string into an array?

**Reply by amreus** [reply]
> Is there a $tw.util to parse a string into an array?

Answer: yes. I found `$tw.util.parseStringArray()` which could be used, but `tiddler` objects have a `getFieldList()` method which seems a better choice.

So I have a proof of concept working with a `parents` field, but no UI will complicate things.  I tend to lose interest after I have something that "works for me."

**Comment by twMat** [comment]
+1

I would like for there to be a custom setting, perhaps like for *default tiddlers* a textarea editor located under *Ctrlpanel>info>advanced* where the names of all fields that should be updated in this action is added by the user. The need to control of this is at least on par with many existing controls in TW. 

Of course, @flibbles [relink plugin](https://github.com/flibbles/tw5-relink) solves this in a much more customizable way.

**Comment by amreus** [comment]
It's kind of odd, but if I change the title of a tiddler to include spaces, the **`parents`** field in other tiddlers field use commas to separate values of the list.  The `list` field however uses `[[` and `]]` to handle spaces in titles.  It's not clear where the `[[` `]]` is added in the code?

**Comment by amreus** [comment]
One way to go might be to add a **`fields`** attribute to the `tm-relink-tiddler` message.  Then each field listed in the `fields` attribute would be handled just like the `tags` and `list` fields are currently.

Another possibility might be to create a tiddler with a module-type of `tiddlerfield`, and use another field such as `relink: yes/no` to include it in relinking events.

**Comment by flibbles** [comment]
Are you... are you talking to yourself?

The method you want for changing arrays back into title lists is `$tw.utils.stringifyList` If you just use `Array.toString()`, that gets you those commas, which is not the TiddlyWiki way.

But like @twMat, the Relink plugin was designed specifically to address your concern. If you're in a situation where you can use it, I recommend you do, otherwise you'd best use the th-rename-tiddler hook or something, which will involve a lot of wheel reinventing.

**Reply by flibbles** [reply]
By the way, I once looked into creating other "array" type fields like "tags" and "list" a while back. There is a module (fieldTypes I think) which is meant for this, but it doesn't really work. It's a chicken-and-the-egg problem.

1. Tiddlywiki reads all tiddlers
2. From those tiddlers, it reads any fieldType modules
3. From that, it learns how to parse other fields as something other than string (array, in this case)
4. but oops, it's already read all tiddlers.

"list" and "tags" are hardcoded inside the boot.js file to be interpreted as arrays. There isn't a nice way to introduce others in a modular way.

...Unless we add a section to boot.js which occurs after modules are process which runs through all tiddlers and converts custom "fieldType" fields to their new types. This way would work, but it does have some downsides.
